src/charm.py

```diff
@@ -859,7 +858,15 @@ def _on_reconcile_runners_action(self, event: ActionEvent) -> None:
                 return
             runner_scaler = self._get_runner_scaler(state)
 
-            delta = runner_scaler.reconcile(state.runner_config.virtual_machines)
+            self.unit.status = MaintenanceStatus("Reconciling runners")
+            try:
+                delta = runner_scaler.reconcile(state.runner_config.virtual_machines)
+            except ReconcileError:
+                logger.exception("Failed to reconcile runners")
+                self.unit.status = MaintenanceStatus("Failed to reconcile runners")
```

Comment:
Are we sure we want to keep the charm in "Maintenance" Status ?

Reply:
I guess your answer is "Yes" we want to keep "Maintenance" status even if there's an error.
What bothers me is that it doesn't fit into the definition of "Maintenance": https://juju.is/docs/juju/status#heading--workload---charm-status

Reply:
Yes, but I'm not sure. I think there is no appropriate status (except error, but this would mean that we shouldn't catch the exception).  ActiveStatus does not fit either because it suggests "Everything is fine".

Reply:
After discussion with @gregory-schiano , we choose ActiveStatus due to this beeing a recoverable error where the workload might still be functional. I think it still not fits "Everything is fine" fully, but might be the better tradeoff.